sort.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of RedisRaft.
+ *
+ * Copyright (c) 2020-2021 Redis Ltd.
+ *
+ * RedisRaft is licensed under the Redis Source Available License (RSAL).
+ */
+
+#include <strings.h>
+
+#include "redisraft.h"
+
+static void replySortedArray(RedisModuleCtx *ctx, RedisModuleCallReply * reply)
+{
+    size_t len = RedisModule_CallReplyLength(reply);
+    if (len == 0) {
+        RedisModule_ReplyWithNullArray(ctx);
+        return;
+    }
+
+    RedisModuleDict * dict = RedisModule_CreateDict(ctx);
+    for(int i=0; i < len; i++) {
+        RedisModuleCallReply * entry = RedisModule_CallReplyArrayElement(reply, i);
+        size_t entry_len;
+        const char * entry_str = RedisModule_CallReplyStringPtr(entry, &entry_len);
+        RedisModule_DictSetC(dict, (char *) entry_str, entry_len, NULL);
+    }
+
+    RedisModule_ReplyWithArray(ctx, len);
+
+    char *key;
+    size_t key_len;
+    void *data;
+    RedisModuleDictIter *iter = RedisModule_DictIteratorStartC(dict, "^", NULL, 0);
+    while((key = RedisModule_DictNextC(iter, &key_len, &data)) != NULL) {
+        RedisModule_ReplyWithStringBuffer(ctx, key, key_len);
+    }
+    RedisModule_DictIteratorStop(iter);
+    RedisModule_FreeDict(ctx, dict);
+}
+
+static void replySortedMap(RedisModuleCtx *ctx, RedisModuleCallReply * reply)
+{
+    size_t len = RedisModule_CallReplyLength(reply);
+    RedisModuleDict * dict = RedisModule_CreateDict(ctx);
+
+    for(int i=0; i < len; i++) {
+        RedisModuleCallReply * key;
+        RedisModuleCallReply * value;
+
+        RedisModule_CallReplyMapElement(reply, i, &key, &value);
+        size_t key_len;
+        const char * key_str = RedisModule_CallReplyStringPtr(key, &key_len);
```

Comment:
```suggestion
    RedisModuleDict *dict = RedisModule_CreateDict(ctx);

    for (size_t i = 0; i < len; i++) {
        RedisModuleCallReply *key;
        RedisModuleCallReply *value;

        RedisModule_CallReplyMapElement(reply, i, &key, &value);
        size_t key_len;
        const char *key_str = RedisModule_CallReplyStringPtr(key, &key_len);
```

Styling / spacing / type correctness.